torch/ao/quantization/_dbr/auto_trace.py

```diff
@@ -114,36 +117,51 @@ def __torch_function__(cls, func, types, args=(), kwargs=None):
             kwargs = kwargs if kwargs else {}
             hook_type = get_torch_function_hook_type(parent_module, func)
 
-            if first_call and hook_type is not HookType.OP_HOOKS:
-                qstate = getattr(parent_module, '_auto_quant_state', None)
-                if qstate:
-                    qstate.add_seen_op_type_without_op_hooks(func)
-
             if hook_type is HookType.OP_HOOKS:
-                qstate = parent_module._auto_quant_state  # type: ignore[attr-defined]
                 fqn = module_id_to_fqn[id(parent_module)] if parent_module else None
+                qstate = parent_module._auto_quant_state  # type: ignore[attr-defined]
                 if not first_call:
                     qstate.validate_cur_op(func)
                 # run "before" hook
                 if first_call:
+                    is_quantizeable_op = True
                     args, kwargs = qstate.first_call_op_prepare_before_hook(
-                        func, args, kwargs, qtensor_id, fqn, parent_module)
+                        func, args, kwargs, qtensor_id, fqn, parent_module,
+                        is_quantizeable_op)
                 else:
                     args, kwargs = qstate.op_prepare_before_hook(
                         func, args, kwargs)
                 # forward
                 output = super().__torch_function__(func, types, args, kwargs)
                 # run "after" hook
                 if first_call:
+                    is_quantizeable_op = True
                     output = qstate.first_call_op_prepare_after_hook(
-                        func, output, args, qtensor_id)
+                        func, output, args, qtensor_id, is_quantizeable_op)
                 else:
                     output = qstate.op_prepare_after_hook(
                         func, output, args, global_op_idx)
                 qstate.mark_cur_op_complete(func)
             else:
```

Comment:
nit: can we add a comment for what this `else` matches to and what this branch means?